### DISCUSSION TITLE
Bind window to desktop layer for running htop etc

### DISCUSSION THREAD

**Discussion by jpmhouston** [ghostty-org/ghostty#3156] [Feature Requests, Ideas]
I know about macOS well but not Linux desktop environments, so this feature request focuses on mac specifically with some guesses about Linux thrown in.

The imagined feature is to take a window one sets up running a continuously program like `htop` and assigns the window the desktop layer beneath all windows and Finder icons. Presumably one would want to have all window chrome omitted and the background transparent.

I currently use the macOS app `Particulars` which draws some system configuration and status onto the desktop layer, replicating this but with the limitless flexibility of whatever can run in a shell is the idea (the app's website at the domain particulars.app seems slow and I don't want to link to it if it's having issues, here's a link to a screenshot elsewhere https://mac-cdn.softpedia.com/screenshots/Particulars_3.jpg and it's in the Mac App Store)

Perhaps interactive could be supported by bringing ghostty.app frontmost, and if other normal windows are open in the app then would switch between them normally to the desktop-bound one. Mouse clicks on the desktop would surely get stolen by the OS and given to the Finder. Perhaps unbinding a window from the desktop, making it a normal window again with full interactivity, and then back, can be a seamless thing one can do anytime for whatever reason.

Restoring this window and restarting its running program when ghostty is relaunched would be ideal.

I'm thinking a window bound to the desktop would be in the on-screen position that the window was beforehand and you could choose to position multiple windows running different programs and bind them to the desktop. Then possibly options about mirroring the same window across all desktop spaces or even screens.

I believe this makes sense on Linux as well, possibly with fewer limitations on mouse input. I think a kitty extension (or whatever they're called) already does something like this and is Linux only.

**Comment by 00-kat** [comment]
> I believe this makes sense on Linux as well

I believe I've seen tools which can do this on Linux without intervention from the app being bound to the desktop.  To do this from Ghostty itself on Wayland, GTK Layer Shells would be required, which is slightly problematic (see also: #4624).

As this discussion is about macOS, I'm not going to comment on this further with regards to Linux, but I just wanted to plop this message in here as a note that it may not make sense or be difficult to support on Linux.
